sequoia/settings/offline_rl/setting.py

```diff
@@ -0,0 +1,110 @@
+import d3rlpy
+import gym
+from gym.wrappers import RecordEpisodeStatistics
+from matplotlib import pyplot as plt
+from sklearn.model_selection import train_test_split
+from typing import ClassVar, List, Tuple, Dict, Any
+
+from sequoia import Results
+from sequoia.settings.base import Setting
+from torch.utils.data import DataLoader
+from sequoia.settings.base import Method
+from dataclasses import dataclass
+
+from simple_parsing.helpers import choice
+
+
+
+@dataclass
+class OfflineRLResults(Results):
+
+    # TODO: Write these methods
+    def summary(self) -> str:
+        return f"Offline RL results: {self.objective_name} = {self.objective}"
+
+    def make_plots(self) -> Dict[str, plt.Figure]:
+        return {}
+
+    def to_log_dict(self, verbose: bool = False) -> Dict[str, Any]:
+        return {}
+
+    # Metrics from online testing
+    test_rewards: list
+    test_episode_length: list
+    test_episode_count: list
+
+    objective_name: ClassVar[str] = "Average Reward"
+
+    @property
+    def objective(self):
+        return sum(self.test_rewards) / len(self.test_rewards)
+
+
+# TODO: smarter way to do this, like a dict { 'offline_datasets_from_d3rlpy': set() } ?
+offline_datasets_from_d3rlpy = {'cartpole-replay', 'cartpole-random'}
+other_datasets = {}
+
+
+@dataclass
+class OfflineRLSetting(Setting):
+    """
+
+    available_datasets: A list of available offline rl datasets
+    dataset: choice of dataset for the current setting
+    val_size: size of the validation set, x out of 1
+
+    create_mask: flag to create binary mask for bootstrapping
+    mask_size: ensemble size for binary mask
+
+    """
+    available_datasets: ClassVar[list] = list(offline_datasets_from_d3rlpy) + list(other_datasets)
+    dataset: str = choice(available_datasets, default="cartpole-replay")
+    val_size: int = 0.2
+
+    # Only d3rlpy uses these params
+    create_mask: bool = False
+    mask_size: int = 1
+
+    def __post_init__(self):
+        # Load d3rlpy offline dataset
+        if self.dataset in offline_datasets_from_d3rlpy:
+            mdp_dataset, self.env = d3rlpy.datasets.get_dataset(self.dataset, self.create_mask, self.mask_size)
+            self.train_dataset, self.valid_dataset = train_test_split(mdp_dataset, test_size=self.val_size)
+
+        # Load other dataset types here
+        elif self.dataset in other_datasets:
```

Comment:
Could you explain what the purpose of this `other_datasets` dict is?
I'm thinking it might be better to just raise a `NotImplementedError` in this case.

Reply:
opted for raising notimplemented error